### PR TITLE
Changes to enable FATES-side running means

### PR DESCRIPTION
**Pull request by rgknox**
### Description of changes

FATES is adding some functions to allow for its own calculation of running means.  This is important because FATES patch data structures are ephemeral and do not have continuity with their associated CLM-side patch/pft indices, and have no analogue for cohorts.  To appropriately perform averages, some time-step information needs to be passed through the interface, and this also removes the usage of the 24-hour average patch-level vegetation temperature from the interface.

### Specific notes

This should be synchronized with: https://github.com/NGEET/fates/pull/724

Contributors other than yourself, if any:

@ckoven  @Qianyuxuan  Claire Zarakas

CTSM Issues Fixed (include github issue #):

(FATES) issue: https://github.com/NGEET/fates/issues/722

Are answers expected to change (and if so in what way)?

Yes, answers should change, subtly, because the new calculation of the 24-hour vegetation temperature factors in ephemeral patches and their fusion. 

Any User Interface Changes (namelist or namelist defaults changes)?

Testing performed, if any:

TBD

(List what testing you did to show your changes worked as expected)
(This can be manual testing or running of the different test suites)
(Documentation on system testing is here: https://github.com/ESCOMP/ctsm/wiki/System-Testing-Guide)
(aux_clm on cheyenne for intel/gnu and izumi for intel/gnu/nag/pgi is the standard for tags on master)

**NOTE: Be sure to check your coding style against the standard
(https://github.com/ESCOMP/ctsm/wiki/CTSM-coding-guidelines) and review
the list of common problems to watch out for
(https://github.com/ESCOMP/CTSM/wiki/List-of-common-problems).**

### CODE REVIEW

**Comment by rgknox** [comment]
Like you say @billsacks .  Immediately following this call to call CLMFatesGlobals(), we step into decompInit_clumps().  This routine needs to know the maximum number of cohorts per grid or column, in order to allocate array spaces.  We need some information on the FATES side prior to this, including NL variables and parameters from the parameter file. I can envision needing more info in the future, for instance when we want to increase the number of patches represented by FATES.  It does not need the timestep in that routine... but I would have to move around a bit of code to make this work.  
I like the idea of keeping the fates globals intializing in one place instead of splitting it, because it seems like it will help us avoid situations where we need two pieces of information that are known at different times in the sequence.  
An alternative is that we could initialize all things just related to timing until after that call to initialize the time manager, which I think happens here:
https://github.com/ESCOMP/CTSM/blob/master/src/main/clm_initializeMod.F90#L303

**Comment by billsacks** [comment]
> Only one test on cheyenne seemed unexpected. SMS_Ld1.nldas2_rnldas2_mnldas2.I2000Ctsm50NwpSpNldas.cheyenne_gnu.clm-default did not compare with base

It looks like this is due to a change in the cprnc version on cheyenne, so now NaNs are being treated as diffs. I'm working with some folks here to try to resolve that issue. Since the only fields that change are fields being sent from datm, I think we can move forward with this tag even with that baseline failure (just note it in the ChangeLog): it seems extraordinarily unlikely to be coming from this branch.

**Comment by ekluzek** [comment]
@rgknox it looks like your branch needs to be updated to the last tag ctsm5.1.dev065 (or perhaps that merge wasn't pushed to your fork)? If the former you likely need to redo your testing, if the later, go ahead and push the latest merge.

Thanks.

**Comment by billsacks** [comment]
I looked into the baseline failure more. @rgknox - It looks like you were unfortunate that these tests ran with an intermediate version of cprnc that now has been replaced. With the latest, cprnc reports that the files are identical, as expected.

**Comment by rgknox** [comment]
Updated tests, after I merged in dev065.  All results are as expected, only BASELINEs differ on FATES tests:

/scratch/cluster/rgknox/tests_1202-204036iz

/glade/scratch/rgknox/tests_1202-203201ch

File: src/main/clm_initializeMod.F90 [hunk 1]
```diff
@@ -222,20 +222,7 @@ subroutine initialize2(ni,nj)
...
-    ! size need.
-    ! Sets:
-    !   fates_maxElementsPerPatch
-    !   fates_maxElementsPerSite (where a site is roughly equivalent to a column)
-    ! (Note: fates_maxELementsPerSite is the critical variable used by CLM
-    ! to allocate space)
-    ! This also sets up various global constants in FATES
-    ! ------------------------------------------------------------------------
-
-    call CLMFatesGlobals()
```
[the remark this answers is not in the document]
Comment (ekluzek) [reply src/main/clm_initializeMod.F90:238]: @rgknox, it looks like the final product doesn't change anything about CLMFatesGlobals. So I'm assuming this means you resolved this question. I'm just checking to make sure we can resolve this conversation.

**Review by ekluzek — APPROVED** [review]

**Pull request merged**